src/test/java/org/openhab/automation/jrule/rules/integration_test/ITJRule.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.automation.jrule.rules.integration_test;
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.openhab.automation.jrule.items.JRuleSwitchItem;
+import org.openhab.automation.jrule.rules.user.TestRules;
+
+/**
+ * The {@link ITJRule}
+ *
+ * @author Robert Delbrück - Initial contribution
+ */
+// Please do not reuse items for testing, create new ones, otherwise every change will hurt sometimes
+public class ITJRule extends JRuleITBase {
+    @Test
+    public void switchItemReceiveCommand() throws IOException {
+        sendCommand(TestRules.ITEM_RECEIVING_COMMAND_SWITCH, JRuleSwitchItem.ON);
+        verifyRuleWasExecuted(TestRules.NAME_SWITCH_ITEM_RECEIVED_ANY_COMMAND);
+        verifyRuleWasExecuted(TestRules.NAME_SWITCH_ITEM_RECEIVED_ANY_UPDATE);
+        verifyRuleWasExecuted(TestRules.NAME_SWITCH_ITEM_CHANGED);
+    }
+
+    @Test
+    public void switchItemReceiveOnCommand() throws IOException {
+        sendCommand(TestRules.ITEM_RECEIVING_COMMAND_SWITCH, JRuleSwitchItem.ON);
+        verifyRuleWasExecuted(TestRules.NAME_SWITCH_ITEM_RECEIVED_ON_COMMAND);
+        verifyRuleWasExecuted(TestRules.NAME_SWITCH_ITEM_RECEIVED_ON_UPDATE);
+        verifyRuleWasExecuted(TestRules.NAME_SWITCH_ITEM_CHANGED_TO_ON);
+    }
+
+    @Test
+    public void invokeMqttAction() throws IOException {
+        sendCommand(TestRules.ITEM_MQTT_ACTION_TRIGGER, JRuleSwitchItem.ON);
+        verifyRuleWasExecuted(TestRules.NAME_INVOKE_MQTT_ACTION);
+        verifyMqttMessageReceived("1313131");
+    }
+
+    @Test
+    public void executeCommandLine() throws IOException, InterruptedException {
+        sendCommand(TestRules.ITEM_COMMANDLINE_EXEC_TRIGGER, JRuleSwitchItem.ON);
+        verifyRuleWasExecuted(TestRules.NAME_EXEC_COMMAND_LINE);
+        verifyFileExist();
+    }
+
+    @Test
+    public void mqttChannelTriggered() throws MqttException {
+        publishMqttMessage("number/state", "123");
+        verifyRuleWasExecuted(TestRules.NAME_MQTT_CHANNEL_TRIGGERED);
```

Comment:
For tests would it not be smart to also verify that outbound events actually make to the eventbus?

Reply:
you mean outbound from the rule method? it's tested if there is some outbound.